src/ucs/stats/stats.c

```diff
@@ -460,6 +465,138 @@ void ucs_stats_node_free(ucs_stats_node_t *node)
+                                .class_name),
+                 "%s", class_name);
+        g_stats_class_id_to_name[g_stats_class_id_to_name_size]
+                .aggregate_sum_class_offset =
+                g_stats_class_id_to_name_last_offset;
+
+        /* Initialize return value */
+        offset = g_stats_class_id_to_name_last_offset;
+
+        /* Return class_id */
+        *class_id = g_stats_class_id_to_name_size;
+
+        /* Update global offset */
+        g_stats_class_id_to_name_last_offset += num_counters_in_class;
+
+        g_stats_class_id_to_name_size++;
+    }
+
+    return offset;
+}
+
+/**
+ * Returns the aggregate-sum of all counters from all interfaces for a compact statistics trace.
+ * Used for Score-P UCX plugin to collect a compact trace of the counters.
+ *
+ * @param [in]  node  Pointer to the statistics node.
+ * @param [in]  sel   Whether or not to recursively iterate through the children.
+ * @param [in]  current_num_counters Current number of counters iterated through (call with 0).
+ * @param [out] ucs_stats_aggregate_sum The aggregate-sum of all counters of each class/type.
+ * @param [in]  max_num_of_stats_counters The number of elements in ucs_stats_aggregate_sum.
+ *
+ * Complexity: O(n)
+ *
+ * Returns: The total number of counters.
+ *
+ * Remarks: The function is recursive.
+ */
+static size_t
+ucs_stats_aggregate_sum_recurs(ucs_stats_node_t *node,
+                               ucs_stats_children_sel_t sel,
+                               size_t current_num_counters,
+                               ucs_stats_counter_s *ucs_stats_aggregate_sum,
+                               size_t max_num_of_stats_counters)
+{
+    ucs_stats_filter_node_t *filter_node = node->filter_node;
+    ucs_stats_node_t *temp_node;
+    ucs_stats_node_t *child;
+    uint32_t counter_index;
+    size_t n_counters = current_num_counters;
+    uint32_t counter_aggregate_sum_array_index;
+    uint32_t counter_output_index;
+    uint32_t num_counters_filtered;
+
+    /* Filter output */
+    counter_output_index = 0;
+
+    num_counters_filtered = ucs_popcount(filter_node->counters_bitmask);
+    counter_aggregate_sum_array_index = 0;
+    ucs_for_each_bit(counter_index, filter_node->counters_bitmask) {
+        ucs_list_for_each(temp_node, &filter_node->type_list_head, type_list) {
+            counter_aggregate_sum_array_index =
+                counter_output_index +
+                ucs_stats_class_id_to_aggregate_sum_base_offset_get(
+                        node->cls->counter_names[counter_index],
+                        &node->cls->class_id, num_counters_filtered);
+
+            if (counter_aggregate_sum_array_index < max_num_of_stats_counters) {
+                ucs_stats_aggregate_sum[counter_aggregate_sum_array_index]
+                        .value += temp_node->counters[counter_index];
+                ucs_stats_aggregate_sum[counter_aggregate_sum_array_index].name =
```

Comment:
do we have re-assign the name+class_name pointer every time you add a value? looks inefficient...

Reply:
Agreed, as we discussed - The function will return the pointer to the const list of class_id's to name and not assign the names each time.

Reply:
Fixed.